src/compiler/program.ts

```diff
@@ -1180,6 +1180,31 @@ namespace ts {
 
             return result;
 
+            function doesStructuralPermitResolutionsReuse(): boolean {
+                switch (structuralIsReused) {
+                    case StructureIsReused.Not: return false;
+                    case StructureIsReused.SafeProjectReferenceModules:
+                        return file.resolvedModules !== undefined &&
+                            isSourceOfProjectReferenceRedirect(file.originalFileName);
+                    case StructureIsReused.SafeModules: return true;
+                    case StructureIsReused.Completely: return true;
+
+                    // It's possible for resolveModuleNamesReusingOldState to be called by
+                    // tryReuseStructureFromOldProgram before structuralIsReused has been defined.
+                    // In this case, the caller (tryReuseStructureFromOldProgram) expects this
+                    // function to reuse resolutions for its checks. Returning true to accommodate
+                    // for that specific scenario.
+                    //
+                    // TODO: Clean this because the above scenario is a bit unexpected. This case

```

Comment:
I think I saw a rule that `TODO` lines need an associated issue, but I'll wait for someone to confirm that this comment is correct before I create that.